### DISCUSSION TITLE
How can I check if my filesystem supports reflinks?

### DISCUSSION THREAD

**Discussion by amoosebitmymom** [treeverse/dvc#10442] [Help] [answered]
Hi
I am using a relatively obscure filesystem in my project. The documentation contains no reference to reflinks. However I want to be absolutely sure. Is there an operation I could simulate using DVC to determine whether or not my filesystem supports reflinks? 
Thanks

**Comment by skshetry** [comment, marked as answer]
`dvc doctor` has a `Cache types` section in the output. If you are not seeing it in the output or has an error link, either you are invoking from a non-dvc repository or have a missing cache directory).
(In that case, I'd `dvc init && mkdir -p .dvc/cache/files/md5`).

**Reply by amoosebitmymom** [reply]
Thanks, that determined it. Is there benefit of using hardlinks over symlinks? Or does it not matter if reflinks aren't available

**Reply by skshetry** [reply]
Practically speaking, I don't think there is much benefit of using hardlink compared to symlink. Hardlink is a bit faster and efficient, but there are limits to how many hardlink you can create in some filesystems. And the performance gap is very small between both of those link types.

You can read some comparisons in [File link types for the DVC cache](https://dvc.org/doc/user-guide/data-management/large-dataset-optimization#file-link-types-for-the-dvc-cache).

Also see https://github.com/iterative/dvc/issues/2459.

**Reply by kmahelona** [reply]
What if `Cache types:` comes back empty? I don't get any errors. While I have a `btrfs` file system, apparently reflinks/deduplication isn't supported because I'm not using SSDs.

Thanks.
